ansible_collections/arista/avd/plugins/plugin_utils/schema/avddataconverter.py

```diff
@@ -1,238 +1,166 @@
-        return instance
-
-    for convert_type in convert_types:
-        if validator.is_type(instance, convert_type):
-            if schema_type in simple_converters:
-                try:
-                    converted_instance = simple_converters[schema_type](instance)
-                except Exception:
-                    # Ignore errors and return original
-                    converted_instance = instance
-                    pass
-                return converted_instance
-            elif convert_type in ["dict", "list"] and schema_type == "list" and "primary_key" in schema:
-                try:
-                    converted_instance = convert_dicts(
-                        instance,
-                        schema["primary_key"],
-                        secondary_key=schema.get("secondary_key"),
-                    )
-                except Exception:
-                    # Ignore errors and return original
-                    converted_instance = instance
-                    pass
-                return converted_instance
-            elif convert_type == "dict" and schema_type == "list":
-                try:
-                    converted_instance = list(instance)
-                except Exception:
-                    # Ignore errors and return original
-                    converted_instance = instance
-                    pass
-                return converted_instance
-    return instance
-
-
-def _is_float(validator, instance):
-    return isinstance(instance, float)
-
-
-"""
-AvdDataConverter is used to convert AVD Data Types based on schema options.
-We have extra type checkers not covered by the AVD_META_SCHEMA (array, boolean etc)
-since the same TypeChecker is used by the converters themselves.
-"""
-if JSONSCHEMA_IMPORT_ERROR or DEEPMERGE_IMPORT_ERROR:
-    AvdDataConverter = None
-else:
-    AvdDataConverter = jsonschema.validators.create(
-        meta_schema=AVD_META_SCHEMA,
-        validators={"$ref": _ref, "items": _items, "keys": _keys},
-        type_checker=jsonschema.TypeChecker(
-            {
-                "any": jsonschema._types.is_any,
-                "array": jsonschema._types.is_array,
-                "boolean": jsonschema._types.is_bool,
-                "integer": jsonschema._types.is_integer,
-                "object": jsonschema._types.is_object,
-                "null": jsonschema._types.is_null,
-                "None": jsonschema._types.is_null,
-                "number": jsonschema._types.is_number,
-                "string": jsonschema._types.is_string,
-                "dict": jsonschema._types.is_object,
-                "str": jsonschema._types.is_string,
-                "bool": jsonschema._types.is_bool,
-                "list": jsonschema._types.is_array,
-                "int": jsonschema._types.is_integer,
-                "float": _is_float,
-            }
-        ),
-    )
+    def __init__(self, avdschema):
+        self._avdschema = avdschema
```

Comment:
It looks like in this case we expect an AvdSchema object when in other similar classes (validator and schemaresolver) we only need a dict. Lets add the type hint here.

I am not quite sure why we need it to be different as it seems the only place we need this kind of object is if the schema is not given to convert_data we resolve the schema. Would it not be enough to instantiate the class with an already resolved schema?

Reply:
Also this would allow for future constants loaded from the schema instance like `ignore keys`

Reply:
Removed type hint again, since this class is imported into AvdSchema and we cannot have a circular import

Reply:
I will add a TODO to refactor this to a resolved schema instead. We have several pending PRs in this area, and I think it needs to wait until all those have merged.

Reply:
resolving but for the future you can have circular import for type hints: https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/